enumerables.rb

```diff
@@ -0,0 +1,173 @@
+    if !block_given?
+      if arg.nil?
+        my_each { |obj| to_change = true unless obj == false || obj.nil? }
+      elsif arg.respond_to?(:to_i)
+        my_each { |obj| to_change = true if obj == arg }
+      elsif arg.is_a?(Regexp)
+        my_each { |obj| to_change = true if obj.match(arg) }
+      elsif arg.respond_to?(:class)
+        my_each { |obj| to_change = true if obj.instance_of? arg }
+      end
+    else
+      my_each { |obj| to_change = true if yield obj }
+    end
+    to_change
+  end
+
+  def my_all?(arg = nil)
+    to_change = true
+    arr = self
+    if !block_given?
+      if arg.nil?
+        my_each { |obj| to_change = false unless arr.include?(true) || obj != false }
+      elsif arg.respond_to?(:to_i)
+        my_each { |obj| to_change = false unless obj == arg }
+      elsif arg.is_a?(Regexp)
+        my_each { |obj| to_change = false unless obj.match(arg) }
+      elsif arg.respond_to?(:class)
+        my_each { |obj| to_change = false unless obj.instance_of? arg }
+      end
+    else
+      my_each { |obj| to_change = false unless yield obj }
+    end
+    to_change
+  end
+
+  def my_none?(arg = nil)
+    to_change = true
+    arr = self
+    if !block_given?
+      if arg.nil?
+        my_each { |obj| to_change = false if arr.include?(true) || obj == true }
+      elsif arg.respond_to?(:to_i)
+        my_each { |obj| to_change = false if obj == arg }
+      elsif arg.is_a?(Regexp)
+        my_each { |obj| to_change = false if obj.match(arg) }
+      elsif arg.respond_to?(:class)
+        my_each { |obj| to_change = false if obj.instance_of? arg }
+      end
+    else
+      my_each { |obj| to_change = false if yield obj }
+    end
+    to_change
+  end
+
+  def my_map
+    arr = self
+    return arr.to_enum unless block_given?
+
+    to_change = []
+    my_each do |obj|
+      to_change.push(yield obj)
+    end
+    to_change
+  end
+
+  def my_count(arg = nil)
+    arr = self
+    total = 0
+    if arg.nil?
+      if block_given?
+        my_each do |item|
+          total += 1 if yield(item)
+        end
+      else
+        total = arr.length
+      end
+    else
+      my_each { |obj| total += 1 if obj == arg }
+    end
+    total
+  end
+
+  def my_inject(start_value = 0, operator = nil)
+    arr = self
+    to_change = ''
+    operator = start_value if start_value.is_a? Symbol
+    start_value = 0 if start_value.is_a? Symbol
+    to_change = 1 if operator == :*
+    to_change = 1 if operator == :/
+    to_change = start_value if start_value != 0
+    if operator.is_a? Symbol
+      case operator
+      when :+
+        arr.my_each { |obj| to_change += obj }
+      when :-
+        arr.my_each { |obj| to_change -= obj }
+      when :*
+        arr.my_each { |obj| to_change *= obj }
+      when :/
+        arr.my_each { |obj| to_change /= obj }
+      end
+    end
+    if block_given?
+      my_each do |obj|
+        to_change = yield to_change, obj
+      end
+    end
+    to_change
+  end
+end
```

Comment:
I highly recommend you to fallow @uryelah  comments regarding this, basically, you've made 2 mistakes the first one would be to check the operator, there is a way more efficient way of doing it but essentially they are not relevant if you build this method properly

the second one but more important is that #my_inject only receives an operator, so the first argument you're asking is never submitted and it's crashing the method. 
maybe what you need is to find the starting of a Range and you could do so with the shift or pop method, I'll let you dig into it ;)

If you need any help with this issue, let me know thought slack on Monday

## Happy Coding

### Daniel Laloush